Frontend/ui-customer/src/Flow.purs

```diff
@@ -2878,4 +2895,288 @@ rideScheduledFlow = do
+          cityName = getCityNameFromCode serviceabilityRes.city
+          (SpecialLocation specialLoc) = fromMaybe HomeScreenData.specialLocation (serviceabilityRes.specialLocation)
+          pickUpPoints = map (\(GatesInfo item) -> {
+              place: item.name,
+              lat  : (item.point)^._lat,
+              lng : (item.point)^._lon,
+              address : item.address,
+              city : Nothing
+            }) specialLoc.gates
+          geoJson = fromMaybe "" serviceabilityRes.geoJson
+          isSpecialZone = (geoJson /= "") && 
+                          (geoJson /= state.data.specialZoneCoordinates) &&
+                          pickUpPoints /= state.data.nearByGates
+          dummyLocInfo = {  lat : Nothing,  lon : Nothing,  placeId : Nothing,  address : "", addressComponents : dummyAddress, city : Nothing}
+          locOnMap = maybe dummyLocInfo identity (state.data.latLonOnMap)
+          updatedState = { lat : fromString lat, lon : fromString lon, placeId : locOnMap.placeId, address : locOnMap.address, addressComponents : locOnMap.addressComponents , city : Just cityName  } 
+      modifyScreenState 
+        $ SearchLocationScreenStateType 
+            (\slsState -> slsState{data{ latLonOnMap = Just updatedState}})
+      if isSpecialZone then 
+        specialLocFlow geoJson pickUpPoints (specialLoc.category) latNum lonNum
+        else 
+          updateLocDetailsFlow state latNum lonNum pickUpPoints cityName
+    
+    SearchLocationController.Reload state -> do 
+      modifyScreenState $ SearchLocationScreenStateType (\_ -> state)
+      searchLocationFlow 
+    SearchLocationController.NoOutput -> pure unit
+    SearchLocationController.SearchPlace searchString state -> searchPlaceFlow searchString state
+    SearchLocationController.SaveFavLoc state savedLoc -> checkRedundantFavLocFlow state savedLoc    
+    SearchLocationController.ConfirmAndSaveFav state -> confirmAndSaveLocFlow state
+    SearchLocationController.PredictionClicked prediction state -> do 
+      modifyScreenState $ SearchLocationScreenStateType (\_ -> state)
+      if state.props.actionType == AddingStopAction then do 
+        void $ lift $ lift $ loaderText (getString STR.LOADING) (getString STR.PLEASE_WAIT_WHILE_IN_PROGRESS)  -- TODO : Handlde Loader in IOS Side
+        void $ lift $ lift $ toggleLoader true
+        let {lat, lon, placeId} = {lat : fromMaybe 0.0 prediction.lat, lon : fromMaybe 0.0 prediction.lon, placeId : prediction.placeId}
+        (GetPlaceNameResp resp) <- getPlaceNameResp (prediction.title <> ", " <> prediction.subTitle) placeId lat lon prediction
+        let (PlaceName placeName) = maybe SearchLocationScreenData.dummyLocationName identity $ head resp
+            (LatLong placeLatLong) = placeName.location
+            {placeLat , placeLon} = {placeLat : placeLatLong.lat, placeLon : placeLatLong.lon}
+        maybe (pure unit) (\ currTextField -> 
```

Comment:
Use helper functions wherever required to make this block readable 

`SearchLocationController.PredictionClicked prediction state -> handlePredictionClick prediction state`

 `(\ currTextField ->  handleCurrTextField currTextField)`